homeassistant/components/roborock/__init__.py

```diff
@@ -47,8 +55,13 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
             translation_key="home_data_fail",
         ) from err
     _LOGGER.debug("Got home data %s", home_data)
+    all_devices: list[HomeDataDevice] = (
+        home_data.devices + home_data.received_devices
+        if entry.options.get(CONF_INCLUDE_SHARED, DEFAULT_INCLUDE_SHARED)
```

Comment:
What's the reason/motivation behind allowing the user to exclude the shared ones?

We usually automatically add all devices, and a user can disable a device.
We should check via the device_registry whether a device is enabled to prevent API calls of deactivated devices.

Reply:
Reason/motivation is just that some users have devices in other homes that they don't care about adding to their HA instance.

It is a feature I have had requested in the past. 

Since the updates are coordinator based - I think disabling the device will stop any api calls (I'm not 100% sure though) - it would probably be good for me to check the device registry in init. I'll add that to my todo list.

As far as this goes. I'm not committed to having the include shared option - so I will defer to your judgement

Reply:
More importantly, IIUC, if a shared device is on a different WiFi network (which I assume they would typically be), the integration cannot control it at all (since this integration primarily uses local control).

Reply:
I never thought about it but I am one of those people.  I have two buildings with vacuums on two different home assistant instances, which share the the same network.

Reply:
We do support cloud connections - but local connections are heavily recommended.

Reply:
Maybe I misunderstood, but everything is local for me.  I have a house and an out building, both buildings have vacuums and separate HA instances. The internet is shared with Ubiquiti AP's that make them appear in the same network space.   I would not want all vacuums to automatically get added to each HA instance.

Reply:
Any update on this @edenhaus if this is a hold up for the PR - I'd be happy to remove it

Reply:
Sorry, I completely missed/forgot about it. I checked with the team, and this should not be part of the option flow.

We should do the following:

> We usually automatically add all devices, and a user can disable a device.
> We should check via the device_registry whether a device is enabled to prevent API calls of deactivated devices if needed.